externs/es6.js

```diff
@@ -776,10 +776,10 @@ function Uint8ClampedArray(length, opt_byteOffset, opt_length) {}
 Uint8ClampedArray.BYTES_PER_ELEMENT;
 
 /**
- * @param {string|!IArrayLike<number>|!Iterable<number>} source
```

Comment:
Thanks for suggestions!

Yep, I worried about unary .from as well; but considered it to be less common than mapping one. Whenever the function is not needed, I would write `new Uint8Array([0, 1, 2])` instead.
Honestly, even when it is needed, `new Uint8Array([0, 1, 2].map(n => 2**n))` feels a bit more natural.
And vice versa: .from is essential for building while ignoring values of "source array", e.g. `Uint8Array.from({length:3}, (_, i) => 2**i)` or even `Uint8Array.from(arrayWithDesiredSize, (_, __) => someGenerator.next().value)`

Of course, since mapless version is more common and we can't "overload" the declaration, I'm reverting it via 1efef958bc816e5f7a86a545472e2a08ff50dbef.
`function(this:S, ?, number): number` looks good. 

> That wouldn't be as important if these methods threw `TypeError` objects for a bad first parameter, but they don't.

Roughly speaking, looks like array is implicitly mapped through something like `Number`, `BigInt` or `n=>Number(n)||0`.
Fortunately, it means that BigInt version *does* throw:
```js
BigUint64Array.from([0]); // TypeError: Cannot convert 0 to a BigInt
BigUint64Array.from([{}]); // SyntaxError: Cannot convert [object Object] to a BigInt
BigUint64Array.from([null]); // TypeError: Cannot convert null to a BigInt
BigUint64Array.from([undefined]); // TypeError: Cannot convert undefined to a BigInt

// out-of-scope but also might happen:
Uint8Array.from([0n]); // TypeError: Cannot convert a BigInt value to a number
Float64Array.from([0n]); // TypeError: Cannot convert a BigInt value to a number
```
Would it be reasonable to not revert it for bigint and just add `@throws {Error}`?

Reply:
> .from is essential for building while ignoring values of "source array", e.g. `Uint8Array.from({length:3}, (_, i) => 2**i)` or even `Uint8Array.from(arrayWithDesiredSize, (_, __) => someGenerator.next().value)`

Those are some odd use-cases. I feel completely comfortable with requiring some fancy type annotations to handle them.

> Of course, since mapless version is more common and we can't "overload" the declaration, I'm reverting it via [1efef95](https://github.com/google/closure-compiler/commit/1efef958bc816e5f7a86a545472e2a08ff50dbef). `function(this:S, ?, number): number` looks good.

Thanks for that.

> Would it be reasonable to not revert it for bigint and just add `@throws {Error}`?

Sorry, but I think it would be weirdly inconsistent to do that.

Reply:
Agreed, reverted that for bigints as well via 1c44581bea75de302c286da63f757c964fb18a5c.

Let me know if squash&rebase is needed.